cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/PreferredLeaderElectionGoal.java

```diff
@@ -51,8 +51,15 @@ public PreferredLeaderElectionGoal(boolean skipUrpDemotion,
     _kafkaCluster = kafkaCluster;
   }
 
+  private void sanityCheckOptimizationOptions(OptimizationOptions optimizationOptions) {
```

Comment:
This is basically the same as the[ one defined in `KafkaAssignerUtils`](https://github.com/linkedin/cruise-control/blob/97e8a66c320eec5a8767e7f75c6f95f5578edd76/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/kafkaassigner/KafkaAssignerUtils.java#L20), maybe we can put this utils in `GoalUtils`

Reply:
Moving this sanity check to `GoalUtils` requires checking whether the given goal satisfies or violates this requirement every time `optimize` is called for any goal. However, we expect this sanity check to be specific to (1) all `KafkaAssigner` goals (hence it is in `KafkaAssignerUtils`), and (2) only to `PreferredLeaderElectionGoal` -- which is contrary to the other goals, executed independent of other goals by itself.

So I don't think we should move this logic to `GoalUtils` as it is irrelevant to other `14 goals`.